executor/show.go

```diff
@@ -595,8 +595,9 @@ func (e *ShowExec) fetchShowCreateTable() error {
 	var hasAutoIncID bool
 	for i, col := range tb.Cols() {
 		fmt.Fprintf(&buf, "  %s %s", escape(col.Name, sqlMode), col.GetTypeDesc())
-		if col.Charset != "binary" {
-			fmt.Fprintf(&buf, " CHARSET %s COLLATE %s", col.Charset, col.Collate)
+		// charset && collate are not shown for generated columns in MySQL
+		if col.Charset != "binary" && !col.IsGenerated() {
```

Comment:
"Because it is generated , charset and collate are not shown." OR "Because it is generated and charset is binary, charset and collate are not shown".